docs/build/building-apps/upgrades/0.52.md

```diff
@@ -0,0 +1,539 @@
+
+Example: 
+
+```go
+app.CircuitKeeper = circuitkeeper.NewKeeper(runtime.NewEnvironment(runtime.NewKVStoreService(keys[circuittypes.StoreKey]), logger.With(log.ModuleKey, "x/circuit")), appCodec, authtypes.NewModuleAddress(govtypes.ModuleName).String(), app.AuthKeeper.AddressCodec())
+```
+
+
+#### Interface Registry
+
+* Replace codec.InterfaceRegistry usage in the HasRegisterInterfaces function with cosmossdk.io/core/registry.InterfaceRegistrar. Same for Amino with cosmossdk.io/core/registry.AminoRegistrar.
+* Example:
+
+```diff
+-func (AppModule) RegisterInterfaces(registry codectypes.InterfaceRegistry) {
++func (AppModule) RegisterInterfaces(registry registry.InterfaceRegistrar) {
+```
+
+#### Amino Codec Registry
+
+* Replace codec.LegacyAmino usage in the HasAminoCodec function with cosmossdk.io/core/registry.AminoRegistrar.
+* Example:
+
+```diff
+-func (AppModule) RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
++func (AppModule) RegisterLegacyAminoCodec(registrar registry.AminoRegistrar) {
+```
+
+### Simulation Changes
+
+* MsgSimulatorFn now returns an error and no longer has sdk.Context; it instead passes in an address.Codec.
+   	* Example:
+
+```diff
+-type MsgSimulatorFn func(r *rand.Rand, ctx sdk.Context, accs []Account) sdk.Msg
++type MsgSimulatorFn func(r *rand.Rand, accs []Account, cdc address.Codec) (sdk.Msg, error)
+```
+
+The interface HasProposalMsgs has been renamed to HasLegacyProposalMsgs, as we've introduced a new simulation framework, simpler and easier to use, named [simsx](https://github.com/cosmos/cosmos-sdk/tree/main/simsx).
+
+### Depinject: Update Usage of Invoke, Provide, Register → appconfig
+
+Previously cosmossdk.io/core/appconfig held functions Invoke, Provide and Register were moved to cosmossdk.io/depinject/appconfig. All modules using dependency injection must update their imports.
+
+```diff
+import {
+- "cosmossdk.io/core/appconfig"
++ "cosmossdk.io/depinject/appconfig"
+}
+```
+
+### Genesis Interfaces Updated
+
+The genesis interfaces in `x/{module_name}/module.go` were modified to avoid Cosmos SDK dependencies.
+
+
+Update your modules’ InitGenesis / ExportGenesis to:
+
+```go
+func (am AppModule) InitGenesis(ctx context.Context, data json.RawMessage) error {
+    // ...
+}
+
+func (am AppModule) ExportGenesis(ctx context.Context) (json.RawMessage, error) {
+    // ...
+}
+```
+
+### Migration to Collections
+
+Modules within the Cosmos SDK were migrated to use [collections](https://github.com/cosmos/cosmos-sdk/tree/main/collections). It is not required to migrate your own modules, but it is recommended to use collections for future modules. 
+
+* If your module or code interacts with updated modules, watch for errors like collections.ErrNotFound instead of older not-found patterns.
```

Comment:
Is there a common previously used error type people may have relied on?  If so, can we call it out?

Reply:
modifed to make more sense,